src/csvcubed/utils/sparql_handler/data_cube_inspector.py

```diff
@@ -207,6 +217,50 @@ def get_code_lists_and_cols(self, csv_url: str) -> CodelistsResult:
 
         return self._codelists_and_cols.get(csv_url, CodelistsResult([], 0))
 
+    def get_dataframe(self, csv_url: str) -> Tuple[pd.DataFrame, List[ValidationError]]:
+        """
+        Get the pandas dataframe for the csv url of the cube wishing to be loaded.
+        """
+        cols = self.get_column_component_info(csv_url)
+        dict_of_types = {}
+        for col in cols:
+            is_attribute_literal = (
+                col.column_type == EndUserColumnType.Attribute
+                and col.column_definition.value_url is None
+            )
+
+            if (
+                col.column_type == EndUserColumnType.Observations
+                or is_attribute_literal
+            ):
+                if col.column_definition.data_type is not None:
```

Comment:
I think this code would read better if it were like this:

```python
if col.column_definition.data_type is None:
   raise ValueError(...)

col_data_type = col.column_definition.data_type.removeprefix(...)
```

That way the assignment of col_data_type isn't hidden away in an if-block.

Reply:
rearranged logic blocks